src/stable_marriage.rs

```diff
@@ -0,0 +1,164 @@
+struct Individual {
+    cnt_match: Option<usize>,
+
+    // For efficiency: most preferred last for men, most preferred first for women
+    is_man: bool,
+    prefs: Vec<usize>,
+}
+
+impl Individual {
+    fn get_pref_idx(&self, idx: usize) -> Option<usize> {
```

Comment:
Similar to the `pop_preferred_if_unmatched`, I guess this could be simplified with simply a `debug_assert!(!self.is_man)`.

That said, if going in this direction, the `Individual` struct could be split into a `Man` and a `Woman` struct for more typechecking… but that's most likely overkill :)